src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_kds.c

```diff
@@ -168,23 +168,33 @@ sk_ecmd2xcmd(struct xocl_dev *xdev, struct ert_packet *ecmd,
 	return 0;
 }
 
-static inline void
-xocl_ctx_to_info(struct drm_xocl_ctx *args, struct kds_ctx_info *info)
+static int
+xocl_initialize_cu_ctx(struct kds_client *client, struct drm_xocl_ctx *args,
```

Comment:
Can we share the code of xocl_initialize_cu_ctx() and zocl_initialize_cu_ctx()?

Reply:
Edge and PCI are using different structure from shim for initializing. We may need to change the shim also to use same structure to pass the argument through ioctl. 
Avoiding to change anything in userspace for this PR.  Will change it later.

Reply:
We can use a intermidiate structure for this purpose, like "struct kds_ctx_info". Change shim is too much.
When initialize cu ctx, you only needs to know cu_idx, cu_domain and flags from "args". You don't need other info. right?

Reply:
Ok. I will make this change